vicodyn/include/cocaine/vicodyn/peer.hpp

```diff
@@ -82,17 +83,26 @@ class peer_t : public std::enable_shared_from_this<peer_t> {
 // thread safe wrapper on map of peers indexed by uuid
 class peers_t {
 public:
-    class app_service_t {
-        using clock_t = std::chrono::steady_clock;
+    using clock_t = std::chrono::steady_clock;
 
+    class app_service_t {
         clock_t::time_point ban_until_;
+        std::unique_ptr<metrics::usts::ewma<clock_t>> timings_ewma_;
 
     public:
-        auto ban(const std::chrono::milliseconds& timeout) -> void;
+        app_service_t(clock_t::duration timings_window);
+
+        auto ban(std::chrono::milliseconds timeout) -> void;
 
         auto banned() const -> bool;
+
+        auto add_request_timing(clock_t::time_point start, clock_t::duration elapsed) -> void;
+
+        auto average_elapsed() const -> clock_t::duration;
```

Comment:
Could you use `double` directly? Is seems to be used only in weighting, where `double` seems appropriate.

Reply:
Fixed